boards/qn9080dk/include/periph_conf.h

```diff
@@ -43,6 +43,28 @@ static const i2c_conf_t i2c_config[] = {
 #define I2C_NUMOF           ARRAY_SIZE(i2c_config)
 /** @} */
 
+/**
+ * @name   SPI configuration
+ * @{
+ */
+static const spi_conf_t spi_config[] = {
+    {
+        .dev            = SPI0,  /* Flexcomm 2 */
+        .copi_pin       = GPIO_PIN(PORT_A, 4),
+        .cipo_pin       = GPIO_PIN(PORT_A, 5),
+        .clk_pin        = GPIO_PIN(PORT_A, 30),
+        .cs_pin         = {
+            GPIO_PIN(PORT_A, 3),  /* MX25R2035F CS# connected here. */
```

Comment:
I didn't look at the `mtd` support yet. Should I add another commit to this PR with the `mtd_spi_nor` config? I tested this interfacing with a SPI device from an application and looking at the signals with the oscilloscope to verify that the driver works as expected.

Reply:
> Should I add another commit to this PR with the `mtd_spi_nor` config?

Yes that would be great. It should be pretty straightforward:

have something like 

```C
static const mtd_spi_nor_params_t _mtd_nor_params = {
    .wait_chip_erase = 50LU * US_PER_SEC,
    .wait_32k_erase = 240LU *US_PER_MS,
    .wait_sector_erase = 40LU * US_PER_MS,
    .wait_chip_wake_up = 35LU * US_PER_MS,
    .clk  = MHZ(32),
    .flag = SPI_NOR_F_SECT_4K | SPI_NOR_F_SECT_32K,
    .spi  = SPI_DEV(0),
    .mode = SPI_MODE_0,
    .cs   = GPIO_PIN(0, 1),
    .wp   = GPIO_PIN(0, 2),
    .hold = GPIO_PIN(0, 3),
    .addr_width = 3,
};

static mtd_spi_nor_t mtd_nor_dev = {
    .base = {
        .driver = &mtd_spi_nor_driver,
        .page_size = 256,
        .pages_per_sector = 16,
    },
    .params = &_mtd_nor_params,
};

mtd_dev_t *mtd0 = (mtd_dev_t *)&mtd_nor_dev;
```

in your `board.c` (verify timings with the [data sheet](https://www.macronix.com/Lists/Datasheet/Attachments/7478/MX25R2035F,%20Wide%20Range,%202Mb,%20v1.6.pdf)) and 

```C
/**
 * @name MTD configuration
 * @{
 */
extern mtd_dev_t *mtd0;
#define MTD_0 mtd0
/** @} */
```

in `board.h`

Then with 

```Make
ifneq (,$(filter mtd,$(USEMODULE)))
  USEMODULE += mtd_spi_nor
endif
```

in `Makefile.dep` you should be good to go.

Reply:
I added an extra commit with the definition of the flash going from the datasheet. I checked all the commands and they are all standard. I noticed there's no way to specify the dummy cycles for the FAST_READ, but then I realized that the driver doesn't even issue .read_fast commands. There's no wake up command in this flash, but any command (any CS# low for 10 ns) wakes it up and the default wake command is a valid one so that's fine. 

However, I can't test this change because I don't have that flash chip on my board. I can flash the mtd_raw and watch the SPI lines, but of course the test fails because I don't have that chip so I can't verify the timings. Do you still want that config?

I was planning to use a QSPI NAND flash connected on the SPIFI interface (a QSPI with a fancier name basically). Is there support for any of this? I didn't see a standard interface for either of these concepts.

Reply:
Hm I should probably push #15300 forward then.

Reply:
Cool! I'll try to implement QSPI based on that API. Do you plan to autosquash the commits? or is the workflow normally to add fixups to the same PR?